bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/html/SingleCharReader.java

```diff
@@ -13,59 +13,15 @@
  *******************************************************************************/
 package org.eclipse.jface.internal.text.html;
 
-import java.io.IOException;
-import java.io.Reader;
+import org.eclipse.text.readers.SingleCharacterReader;
 
 
 /**
  * <p>
  * Moved into this package from <code>org.eclipse.jface.internal.text.revisions</code>.</p>
+ *
+ * @deprecated Please use SingleCharacterReader
  */
-public abstract class SingleCharReader extends Reader {
-
-	/**
-	 * @see Reader#read()
-	 */
-	@Override
-	public abstract int read() throws IOException;
-
-	/**
-	 * @see Reader#read(char[],int,int)
-	 */
-	@Override
-	public int read(char cbuf[], int off, int len) throws IOException {
-		int end= off + len;
-		for (int i= off; i < end; i++) {
-			int ch= read();
-			if (ch == -1) {
-				if (i == off)
-					return -1;
-				return i - off;
-			}
-			cbuf[i]= (char)ch;
-		}
-		return len;
-	}
-
-	/**
-	 * @see Reader#ready()
-	 */
-	@Override
-	public boolean ready() throws IOException {
-		return true;
-	}
+public abstract class SingleCharReader extends SingleCharacterReader {
```

Comment:
Let's add the official `@Deprecated` java annotation as well.
Should we plan and announce removal?